README.md

```diff
@@ -54,34 +56,92 @@ source your_virtual_env/bin/activate
 
 ## :rocket: Usage
 
-### From script
+### Link and Joint Modifier
+
+These modifiers change the properties of the specific link/joint they are connected to. 
 
 ```python
 from urdfModifiers.core.linkModifier import LinkModifier
 from urdfModifiers.core.jointModifier import JointModifier
 from urdfModifiers.core.modification import Modification
 from urdfModifiers.utils import *
+from urdfModifiers.geometry import *
 
 urdf_path ="./models/stickBot/model.urdf"
 output_file = "./models/stickBotModified.urdf"
 dummy_file = 'no_gazebo_plugins.urdf'
+# Extract the <gazebo> tags from the urdf, as they collide with the library
 robot, gazebo_plugin_text = utils.load_robot_and_gazebo_plugins(urdf_path,dummy_file)
 
-modifications = Modification()
-modifications.add_dimension(0.2, absolute=True) # Makes the link 0.2m long
-modifications.add_density(2.0, absolute=False) # Doubles the link's density
+# Create a link modifier by specifying link name and robot
+# axis parameter is necessary for dimension and position modifications
+link_modifier = LinkModifier.from_name('r_upper_arm', robot, axis=geometry.Side.Z)
+# Same applies for JointModifier
+joint_modifier = JointModifier.from_name('r_elbow', robot, axis=geometry.Side.X)
+
+# Create a new Modification instance
+link_modifications = Modification()
+# Make the link 0.2m long
+link_modifications.add_dimension(0.2, absolute=True) 
+# Double the link's density
+link_modifications.add_density(2.0, absolute=False) 
+
+# Create a new Modification instance 
+joint_modifications = Modification()
+# Move the origin of the joint to 0.4 in the modifier's axis
+joint_modifications.add_position(0.4, absolute=True) 
+
+# Apply the modifications
+link_modifier.modify(link_modifications)
+joint_modifier.modify(joint_modifications)
+
+# Write URDF to a new file, also adding back the previously removed <gazebo> tags                
+utils.write_urdf_to_file(robot, output_file, gazebo_plugin_text)  
+```
 
+### FixedOffsetModifier
 
-modifiers = [LinkModifier.from_name('r_upper_arm',robot, 0.022),
-                JointModifier.from_name('r_elbow',robot, 0.0344)]
-                
-for item in modifiers:
-    item.modify(modifications)
-utils.write_urdf_to_file(robot, output_file, gazebo_plugin_text)  
+This modifier is meant to be connected to a link. It will perform the same modifications as a simple `LinkModifier`. 
+
+However, for **dimension** modifications it will also adjust the origin of both the link and its child joint (that is, the joint that comes immediately after this link in the URDF tree), so that the distance between the beginning of the link and its parent joint (**parent offset**), as well as the distance between the end of the link and its child joint (**child offset**) are preserved. You can see a diagram of this modification in the following image:
+
+![FixedOffset Explanation](https://user-images.githubusercontent.com/31577366/155292943-dda056cb-a72c-4168-b283-85475d1d5e43.png)
```

Comment:
Do you have the source of this image if someone in the future needs to modify it? Just attaching it to the PR is probably enough.

Reply:
Hi Silvio, I added it as part of the repo in [this](https://github.com/icub-tech-iit/urdf-modifiers/commit/a418aac90e24330cc58e67acfd1a1b7d8a2ddf13) commit